pkg/provision/azure.go

```diff
@@ -0,0 +1,368 @@
+package provision
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2017-12-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2017-05-10/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/google/uuid"
+	"github.com/sethvargo/go-password/password"
+	"log"
+	"os"
+)
+
+const AzureStatusSucceeded = "Succeeded"
+
+type AzureProvisioner struct {
+	subscriptionId    string
+	resourceGroupName string
+	deploymentName    string
+	authorizer        autorest.Authorizer
+	ctx               context.Context
+}
+
+func NewAzureProvisioner(subscriptionId, authFile string) (*AzureProvisioner, error) {
+	err := os.Setenv("AZURE_AUTH_LOCATION", authFile)
+	if err != nil {
+		return nil, err
+	}
+	authorizer, err := auth.NewAuthorizerFromFile(azure.PublicCloud.ResourceManagerEndpoint)
+	ctx := context.Background()
+	return &AzureProvisioner{
+		subscriptionId: subscriptionId,
+		authorizer:     authorizer,
+		ctx:            ctx,
+	}, err
+}
+
+// Provision provisions a new Azure instance as an exit node
+func (p *AzureProvisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
+
+	log.Printf("Provisioning host with Azure\n")
+
+	p.resourceGroupName = "inlets-" + host.Name
+	p.deploymentName = "inlets-deploy-" + uuid.New().String()
+
+	log.Printf("Creating resource group %s", p.resourceGroupName)
```

Comment:
Can you add a `Debug` true/false to the constructor for the provisioner? So that we can turn these messages off when needed.

Reply:
Currently there seems no debug mode on the top level and no command line flag to switch on/off. Introducing debug mode would need to modify the interface, hence will need to modify all the provisioners. 

Or we introduce some kind of global setting variable that is accessible from everywhere? (Maybe not a good idea though)